### ISSUE TITLE
Microservice Platform

### ISSUE THREAD

**Issue by eljam** [Tolerance/Tolerance#53]
I open an new issue related to the discussion in #52 PR.

I agree with you about the php ecosystem, we lack this kind of initiative as there is a lot of how to do an api rest application but we don't have anything to make them talk at scale. 

It could be interesting to have a sort of "microservice platform" that can leverage all features from symfony and will be based on small microservice components if you love laravel or code igniter :trollface: :)

Microservice Plaftorm (microservice chassis framework)

- Endpoint (they all use RPC messaging pattern, i don't know if it's the best solution)
- Authentication system
- log(monolog multiple output)
- monitoring/health check
- metrics
- circuit breaker
- rate limiter (you already have one)
- transport (you already have one)
- service discovery (client side discovery or server side discovery) use it with etcd or another key/value store.
- router/load balancing

I'm not a master in microservices, i just read some articles a long time ago :).

What do you think, this looks completely wrong ?

**Comment by ProPheT777** [comment]
IMO, most of points are already solved with other applications that doing well.

More over we are in PHP, so we have some limitation (especially for load balancing, routing) by the nature of the language so I prefer see an integration with these kind of tools (I have traefik in my head for example) and interact with them via php than create a new implementation to do the same job. 

But I like the idea to have a PHP (platform / tools) that can (aggregate / make easier) all this kind of features and bring it to our application level and bring a clear idea of what you should achieve to do micro services in the right way.

**Comment by eljam** [comment]
Yes i agree with do not reinvent the wheel. I was thinking about something like sylius where you can use own framework or the plug and play version based on symfony.

**Comment by ProPheT777** [comment]
Yes we share the vision, expect about symfony (I'm symfony enthusiast) but I prefer reuse component that I need and make my own bootstrap than use a fat standard edition to doing micro services.

**Comment by eljam** [comment]
Indeed the big fat standard edition is maybe a bit too much even if you want a plug and play solution. Bridge should be enough to begin with another framework.
